resources/windows/wix/bootsnap.vbs

```diff
@@ -0,0 +1,158 @@
+  else
+    ExecuteCommand = True
+  End If
+End Function
+
+Function GetRubyDirectory(RootDirectory)
+  GetRubyDirectory = ""
+  ' Find a ruby environment with the PDK gem
+  Dim RubyFolder : Set RubyFolder = fso.GetFolder(RootDirectory)
+  For Each RubyVerSubfolder in RubyFolder.SubFolders
+    Dim RubyFullVersion : RubyFullVersion = RubyVerSubfolder.Name
+    Dim arrRubyVersion : arrRubyVersion = Split(RubyFullVersion, ".")
+    Dim RubyMinorVersion : RubyMinorVersion = arrRubyVersion(0) + "." + arrRubyVersion(1) + ".0"
+    Dim GemFolderPath : GemFolderPath = RubyVerSubfolder.Path + "\lib\ruby\gems\" + RubyMinorVersion + "\gems"
+
+    if fso.FolderExists(GemFolderPath) Then
+      Dim FoundPDK : FoundPDK = false
+      For Each GemFolder in fso.GetFolder(GemFolderPath).SubFolders
+        FoundPDK = FoundPDK OR Left(GemFolder.Name,4) = "pdk-"
+      Next
+      If FoundPDK Then
+        GetRubyDirectory = RubyFullVersion
+      End If
+    End If
+  Next
+End Function
+
+' Mainline
+Function RunRubyCommand(command)
+  Log "InstallDir is " + InstallDir, false
+
+  ' Based on equivalent PowerShell script at;
+  ' https://github.com/puppetlabs/pdk-vanagon/blob/07a4ee7c29ba630ffbec6d87388688b6de90fbfd/resources/files/windows/PuppetDevelopmentKit/PuppetDevelopmentKit.psm1
+  Dim BOLT_BASEDIR
+  Dim RUBY_DIR
+  Dim SSL_CERT_FILE
+  Dim SSL_CERT_DIR
+
+  BOLT_BASEDIR = fso.GetFolder(InstallDir).ShortPath
+
+  Log "BOLT_BASEDIR is " + BOLT_BASEDIR, false
+  RUBY_DIR = BOLT_BASEDIR + "\lib\ruby\"
+  SSL_CERT_FILE = BOLT_BASEDIR + "\ssl\cert.pem"
+  SSL_CERT_DIR = BOLT_BASEDIR + "\ssl\certs"
+
+  Log "BOLT_BASEDIR is " + BOLT_BASEDIR, false
+  Log "RUBY_DIR is " + RUBY_DIR, false
+  Log "SSL_CERT_FILE is " + SSL_CERT_FILE, false
+  Log "SSL_CERT_DIR is " + SSL_CERT_DIR, false
+
+  Dim RubyPath : RubyPath = BOLT_BASEDIR + "\bin\ruby.exe"
+  Dim BundlePath : BundlePath = BOLT_BASEDIR + "\bin\bundle"
+  Dim ProcessEnv : Set ProcessEnv = wshShell.Environment( "PROCESS" )
+
+  Log "Creating process level environment variables...", false
+  ProcessEnv("BOLT_BASEDIR") = BOLT_BASEDIR
+  ProcessEnv("RUBY_DIR") = RUBY_DIR
+  ProcessEnv("SSL_CERT_FILE") = SSL_CERT_FILE
+  ProcessEnv("SSL_CERT_DIR") = SSL_CERT_DIR
+  ProcessEnv("PDK_DEBUG") = "True"
+  Dim NewEnvPath : NewEnvPath = BOLT_BASEDIR + "\bin;" + ProcessEnv("PATH")
+  ProcessEnv("PATH") = NewEnvPath
+
+  ' Note Returning values back to the MSI Engine only works with Binary type Custom Actions
+  ' if ExecuteCommand("""" & RubyPath & """ -S -- """ & ExtractScript & """") Then
+  if ExecuteCommand(command) Then
+    Log "Completed with success", false
+    RunRubyCommand = IDOK
+  Else
+    Log "Completed with error", true
+    RunRubyCommand = IDABORT
+  End If
+End Function
+
+Function RunBootsnap()
+  Dim BOLT_BASEDIR
+  Dim cmd
+  BOLT_BASEDIR = fso.GetFolder(InstallDir).ShortPath
+  cmd = "bootsnap precompile '" + BOLT_BASEDIR + "' --cache-dir c:/ProgramData/Puppetlabs/bolt"
```

Comment:
This needs to be changed to the an environment variable that points to the drive programdata is on